test/unit/org/apache/cassandra/index/sai/cql/LuceneAnalyzerTest.java

```diff
@@ -584,22 +626,28 @@ public void testTokenizer() throws Throwable
     }
 
     @Test
-    public void testAnalyzerMatchesAndEqualityFailForConjunction() throws Throwable
+    public void testMixedAnalyzerMatchesAndEquality() // there are more detailed tests in AnalyzerEqOperatorSupportTest
     {
         createTable("CREATE TABLE %s (id text PRIMARY KEY, val text)");
 
-        createIndex("CREATE CUSTOM INDEX ON %s(val) USING 'StorageAttachedIndex' WITH OPTIONS = {'index_analyzer':'\n" +
-                    "\t{\"tokenizer\":{\"name\":\"whitespace\"}," +
-                    "\t\"filters\":[{\"name\":\"porterstem\"}]}'}");
+        createIndex("CREATE CUSTOM INDEX ON %s(val) USING 'StorageAttachedIndex' WITH OPTIONS = {" +
+                    "'index_analyzer':'{\n" +
+                    "   \"tokenizer\":{\"name\":\"whitespace\"}," +
+                    "   \"filters\":[{\"name\":\"porterstem\"}]" +
+                    "}'," +
+                    "'equals_behaviour_when_analyzed': 'MATCH'}");
 
         waitForIndexQueryable();
 
         execute("INSERT INTO %s (id, val) VALUES ('1', 'the queries test')");
 
-        assertThatThrownBy(() -> execute("SELECT * FROM %s WHERE val : 'queries' AND val : 'the' AND val = 'the queries test'"))
-        .isInstanceOf(UnsupportedOperationException.class);
-        assertThatThrownBy(() -> execute("SELECT * FROM %s WHERE val : 'queries' AND val : 'the' AND val = 'the queries test' ALLOW FILTERING"))
-        .isInstanceOf(UnsupportedOperationException.class);
```

Comment:
Hmmm, so the intention was not to support mixed `:` and `=`, and we changed it here? How hard is it to make it unsupported again?